### PR TITLE
Add authentication servlet and logic to configure header logic status

### PR DESCRIPTION
**Pull request by jyx11011**
This PR was approved earlier but it was not merged to master.

### CODE REVIEW

**Comment by limli** [comment]
can you `git merge master`? When I checkout this branch, it can't be compiled because of a syntax error in the `web.xml` file

**Review by limli — COMMENTED** [review]
looks good to me

File: src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -1,28 +1,93 @@
+<<<<<<< HEAD
```
Comment (anuragshirolkar) [inline src/main/webapp/WEB-INF/web.xml]: Looks like there's merge conflict.

**Review by anuragshirolkar — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/web.xml (hunk 1 above)
Comment (SuyashLakhotia) [inline src/main/webapp/WEB-INF/web.xml]: Seems like there are unresolved merge conflicts?

**Review by anuragshirolkar — APPROVED** [review]

**Pull request merged**